h2o-persist-hdfs/src/main/java/water/persist/security/HdfsDelegationTokenRefresher.java

```diff
@@ -0,0 +1,215 @@
+package water.persist.security;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import org.apache.hadoop.security.token.TokenIdentifier;
+import org.apache.hadoop.security.token.delegation.AbstractDelegationTokenIdentifier;
+import water.H2O;
+import water.MRTask;
+import water.Paxos;
+import water.persist.PersistHdfs;
+import water.util.BinaryFileTransfer;
+import water.util.FileUtils;
+
+import java.io.*;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class HdfsDelegationTokenRefresher implements Runnable {
+
+    public static final String H2O_AUTH_USER = "h2o.auth.user";
+    public static final String H2O_AUTH_PRINCIPAL = "h2o.auth.principal";
+    public static final String H2O_AUTH_KEYTAB = "h2o.auth.keytab";
+    public static final String H2O_AUTH_TOKEN_REFRESHER_ENABLED = "h2o.auth.tokenRefresher.enabled";
+    public static final String H2O_AUTH_TOKEN_REFRESHER_INTERVAL_RATIO = "h2o.auth.tokenRefresher.intervalRatio";
```

Comment:
Is this going to come from some H2O driver parameter? I can't see the origin of the value right now.

Reply:
This is never populated, h2odriver has a generic mechanism of passing properties and adding them to the configuration. This way user can change the property without the driver really doing anything for it.

Reply:
(it is great you catch these things though!)